staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -69,7 +69,7 @@ type CachedDiscoveryInterface interface {
 	// and doesn't make a lot of sense.
 	Fresh() bool
 	// Invalidate enforces that no cached data is used in the future that is older than the current time.
-	Invalidate()
```

Comment:
I think the memcache implementation of this does not match the documented meaning of this function. That seems like the right thing to address first

Reply:
I'm opening another issue to refactor the memcache implementation. I'll link it here.

In this PR, I just want to make a very narrow-scoped fix.

Reply:
I wanted to rename the method to `Refresh()`, which would match the current implementation of the memcache client more. But unfortunately it doesn't match the disk-cache implementation.

Reply:
those two operations seem distinct and independently useful. Not sure I'd want to trade one for the other

Reply:
Yes, we can add both to the interface, noted in #68865, which I will work on in 1.13. The PR will not be small enough to cherrypick to 1.12.

This PR only aims to fix #68735. @wojtek-t do we still want to fix #68735 in 1.12.1? If not, I can close this PR.

Reply:
I think we should (I took a look in the morning and this PR looks reasonable to me, but I'm not very familiar with this part of code).

But I think would leave the ultimate decision to someone from apimachinery [ @lavalamp ? @deads2k ?]  (I didn't see this error in last 4 runs, so it's not that frequent).